clients/src/main/java/org/apache/kafka/common/security/authenticator/DefaultKafkaPrincipalBuilder.java

```diff
@@ -44,16 +45,19 @@
  * class applies {@link org.apache.kafka.common.security.kerberos.KerberosShortNamer} to transform
  * the name.
  *
- * NOTE: This is an internal class and can change without notice
+ * NOTE: This is an internal class and can change without notice. Unlike normal implementations
+ * of {@link KafkaPrincipalBuilder}, there is no default no-arg constructor since this class
+ * must adapt implementations of the older {@link PrincipalBuilder} interface.
  */
-public class DefaultKafkaPrincipalBuilder implements KafkaPrincipalBuilder, AutoCloseable {
+public class DefaultKafkaPrincipalBuilder implements KafkaPrincipalBuilder, Closeable {
     private final PrincipalBuilder oldPrincipalBuilder;
     private final Authenticator authenticator;
     private final TransportLayer transportLayer;
     private final KerberosShortNamer kerberosShortNamer;
 
     /**
-     * Construct a new instance.
+     * Construct a new instance which wraps an instance of the older {@link PrincipalBuilder}.
+     *
      * @param authenticator The authenticator in use
      * @param transportLayer The underlying transport layer
      * @param oldPrincipalBuilder Instance of {@link PrincipalBuilder} or null if none is configured
```

Comment:
Needs to be updated to match the implementation. Also, would it be better to expose a static factory method `fromOldPrincipalBuilder` or something? The constructor would then become private. Seems a bit clearer.